internal/logger/io_sink.go

```diff
@@ -27,71 +29,26 @@ func NewIOSink(out io.Writer) *IOSink {
 	}
 }
 
-func logCommandMessageStarted(log *log.Logger, kvMap map[string]interface{}) {
-	format := "Command %q started on database %q using a connection with " +
-		"server-generated ID %d to %s:%d. The requestID is %d and " +
-		"the operation ID is %d. Command: %s"
-
-	log.Printf(format,
-		kvMap["commandName"],
-		kvMap["databaseName"],
-		kvMap["serverConnectionId"],
-		kvMap["serverHost"],
-		kvMap["serverPort"],
-		kvMap["requestId"],
-		kvMap["operationId"],
-		kvMap["command"])
-
-}
-
-func logCommandMessageSucceeded(log *log.Logger, kvMap map[string]interface{}) {
-	format := "Command %q succeeded in %d ms using server-generated ID " +
-		"%d to %s:%d. The requestID is %d and the operation ID is " +
-		"%d. Command reply: %s"
-
-	log.Printf(format,
-		kvMap["commandName"],
-		kvMap["duration"],
-		kvMap["serverConnectionId"],
-		kvMap["serverHost"],
-		kvMap["serverPort"],
-		kvMap["requestId"],
-		kvMap["operationId"],
-		kvMap["reply"])
-}
-
-func logCommandMessageFailed(log *log.Logger, kvMap map[string]interface{}) {
-	format := "Command %q failed in %d ms using a connection with " +
-		"server-generated ID %d to %s:%d. The requestID is %d and " +
-		"the operation ID is %d. Error: %s"
-
-	log.Printf(format,
-		kvMap["commandName"],
-		kvMap["duration"],
-		kvMap["serverConnectionID"],
-		kvMap["serverHost"],
-		kvMap["serverPort"],
-		kvMap["requestId"],
-		kvMap["operationId"],
-		kvMap["failure"])
-}
-
+// Info will write the provided message and key-value pairs to the io.Writer
+// as extended JSON.
 func (osSink *IOSink) Info(_ int, msg string, keysAndValues ...interface{}) {
 	kvMap := make(map[string]interface{})
+	kvMap[KeyMessage] = msg
+
 	for i := 0; i < len(keysAndValues); i += 2 {
 		kvMap[keysAndValues[i].(string)] = keysAndValues[i+1]
 	}
 
-	switch msg {
-	case CommandStarted:
-		logCommandMessageStarted(osSink.log, kvMap)
-	case CommandSucceeded:
-		logCommandMessageSucceeded(osSink.log, kvMap)
-	case CommandFailed:
-		logCommandMessageFailed(osSink.log, kvMap)
+	kvBytes, err := bson.MarshalExtJSON(kvMap, false, false)
+	if err != nil {
+		panic(err)
 	}
+
+	osSink.log.Println(string(kvBytes))
 }
 
+// Error will write the provided error and key-value pairs to the io.Writer
+// as extended JSON.
 func (osSink *IOSink) Error(err error, msg string, kv ...interface{}) {
 	osSink.Info(0, msg, kv...)
```

Comment:
I just realized `Error` doesn't do anything with `err`. We should include `err` in the `keysAndValues` before passing them to `Info`.